Library/Formula/boost.rb

```diff
@@ -20,6 +20,10 @@ class Boost < Formula
   option "without-single", "Disable building single-threading variant"
   option "without-static", "Disable building static library variant"
   option "with-mpi", "Build with MPI support"
+  option "with-locale", "Build with locale library"
+  option "with-filesystem", "Build with filesystem library"
+  option "with-program_options", "Build with program_options library"
+  option "with-regex", "Build with regex library"
```

Comment:
Why we need these options? It seems to be me that we already build these features by default:
```
$ brew ls boost | grep -oE 'libboost_\w+' | cut -d _ -f 2 | uniq | sort
atomic
chrono
container
context
coroutine
date
exception
filesystem
graph
iostreams
locale
log
math
prg
program
random
regex
serialization
signals
system
test
thread
timer
type
unit
wave
wserialization
```